Test on more versions of Python

maresb opened the pull request:
<!-- Hello! Thanks for submitting a PR! To help make things go a bit more
     smoothly, we would appreciate it if you follow this template. -->

### Description
Dual to #562 let's check which Python versions work.
<!-- Good things to put here include:
       - reasons for the change (please link any relevant issues!),
       - any noteworthy (or hacky) choices to be aware of,
       - or what the problem resolved here looked like. -->



<!-- Just as a reminder, everyone in all conda org spaces (including PRs)
     must follow the Conda Org Code of Conduct (link below).

     Finally, once again, thanks for your time and effort. If you have any
     feedback in regards to your experience contributing here, please
     let us know!

     Helpful links:
       - Conda Org COC: https://github.com/conda/.github/blob/main/CODE_OF_CONDUCT.md
       - Contributing docs: https://github.com/conda/.github/blob/main/CONTRIBUTING.md -->

maresb commented:
Ok, conda-recipe-manager fails unless Python >=3.11

maresb commented:
Probably the right thing to do is conditionally install conda-recipe-manager on Python >=3.11 and then conditionally run only the tests that actually need it.

marcelotrevisani commented:
yeah, that is due to the fact that ``conda-recipe-manager`` constrain python
but that is a optional dependency, we can indeed run for older python versions without it

maresb commented:
It looks like there is no coverage yet for `--use-v1-format` so the tests already pass without `conda-recipe-manager`.

maresb commented:
Oops, seems to be hitting https://github.com/mamba-org/setup-micromamba/issues/225

maresb commented:
@marcelotrevisani, what do you think of these changes?

Tests are now passing down to Python 3.8. In order to get there I had to `from __future__ import annotations` to some files, tweak a context manager call, and run `ruff check --fix`.

I removed `conda-recipe-manager` from the `environment.yml`, but in the CI I conditionally install it when the Python version is >=3.11.

Finally, I switched from `setup-miniconda` to `setup-micromamba`.

marcelotrevisani commented on environment.yaml:
```diff
@@ -29,4 +29,4 @@ dependencies:
   - beautifulsoup4
   - semver >=3.0.0,<4.0.0
-  - conda-recipe-manager >=0.2.0
+  - micromamba
```
do we need micromamba here? 😬

marcelotrevisani commented on grayskull/utils.py:
```diff
@@ -244,7 +244,15 @@ def upgrade_v0_recipe_to_v1(recipe_path: Path) -> None:
+        raise ImportError(
+            "Please install conda-recipe-manager from conda-forge to enable "
+            "support for the V1 format. (Note that Python >=3.11 is required.)"
+        ) from e
```
cool, thanks for that

marcelotrevisani commented:
hey, thanks again for this PR, I just added one comment about micromamba in the environment.yaml, do we need it there?

maresb commented:
Ah, very good point!!! Now that I switched the CI over to `setup-micromamba` I don't think we need it since I needed it specifically for that CI.

I just rebased to remove the commit where I added it. (See the rebase diff with the "Compare" button or equivalently [here](https://github.com/conda/grayskull/compare/a078d44870a0d6ea688972ff3cbe2bdacda823b0..137d60c4c172ad40e74eb7610c9be9c37c8f12be).)

maresb commented:
@marcelotrevisani, yep, it's working. The relevant part of CI has completed.
